[CI] MixedClusterEsqlSpecIT test {union_types.MultiIndexSortIpStringEval ASYNC} failing

**Build Scans:**
- [elasticsearch-periodic #7922 / openjdk23_8.19.0_java-matrix-bwc](https://gradle-enterprise.elastic.co/s/a3xhpgxoopm2e)
- [elasticsearch-pull-request #69454 / 8.19.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/6fm3sung2stum)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:mixed-cluster:v8.19.0#javaRestTest" -Dtests.class="org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT" -Dtests.method="test {union_types.MultiIndexSortIpStringEval ASYNC}" -Dtests.seed=C7CEC9DEFEB0EEC1 -Dtests.bwc=true -Dtests.locale=mt-MT -Dtests.timezone=Israel -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Bunion_types.MultiIndexSortIpStringEval%20ASYNC%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: expected:<[{name=@timestamp, type=date}, {name=client_ip, type=unsupported, original_types=[ip, keyword], suggested_cast=keyword}, {name=event_duration, type=long}, {name=message, type=keyword}, {name=client_ip_as_ip, type=ip}]> but was:<[{name=@timestamp, type=date}, {name=client_ip, type=unsupported}, {name=event_duration, type=long}, {name=message, type=keyword}, {name=client_ip_as_ip, type=ip}]>
```

**Issue Reasons:**
- [main] 2 failures in test test {union_types.MultiIndexSortIpStringEval ASYNC} (0.7% fail rate in 306 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -432,9 +432,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT
   method: test
   issue: https://github.com/elastic/elasticsearch/issues/127536
-- class: org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT
-  method: test {union_types.MultiIndexSortIpStringEval ASYNC}
-  issue: https://github.com/elastic/elasticsearch/issues/127537
 - class: org.elasticsearch.xpack.inference.action.filter.ShardBulkInferenceActionFilterIT
   method: testRestart {p0=false p1=false}
   issue: https://github.com/elastic/elasticsearch/issues/127592

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RestEsqlTestCase.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RestEsqlTestCase.java
@@ -1289,6 +1289,7 @@ public abstract class RestEsqlTestCase extends ESRestTestCase {
         String id = (String) json.get("id");
 
         var supportsAsyncHeaders = clusterHasCapability("POST", "/_query", List.of(), List.of("async_query_status_headers")).orElse(false);
+        var supportsSuggestedCast = clusterHasCapability("POST", "/_query", List.of(), List.of("suggested_cast")).orElse(false);
 
         if (id == null) {
             // no id returned from an async call, must have completed immediately and without keep_on_completion
@@ -1334,13 +1335,39 @@ public abstract class RestEsqlTestCase extends ESRestTestCase {
 
         // assert initial contents, if any, are the same as async get contents
         if (initialColumns != null) {
-            assertEquals(initialColumns, result.get("columns"));
+            if (supportsSuggestedCast == false) {
+                assertEquals(
+                    removeOriginalTypesAndSuggestedCast(initialColumns),
+                    removeOriginalTypesAndSuggestedCast(result.get("columns"))
+                );
+            } else {
+                assertEquals(initialColumns, result.get("columns"));
+            }
             assertEquals(initialValues, result.get("values"));
         }
 
         assertWarnings(response, assertWarnings);
         assertDeletable(id);
         return removeAsyncProperties(result);
+    }
+
+    private static Object removeOriginalTypesAndSuggestedCast(Object response) {
+        if (response instanceof ArrayList<?> columns) {
+            var newColumns = new ArrayList<>();
+            for (var column : columns) {
+                if (column instanceof Map<?, ?> columnMap) {
+                    var newMap = new HashMap<>(columnMap);
+                    newMap.remove("original_types");
+                    newMap.remove("suggested_cast");
+                    newColumns.add(newMap);
+                } else {
+                    newColumns.add(column);
+                }
+            }
+            return newColumns;
+        } else {
+            return response;
+        }
     }
 
     public void testAsyncGetWithoutContentType() throws IOException {

